velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -923,7 +923,9 @@ struct DateTruncFunction : public TimestampWithTimezoneSupport<T> {
     auto dateTime = getDateTime(timestamp, timeZone_);
     adjustDateTime(dateTime, unit);
 
-    result = Timestamp(timegm(&dateTime), 0);
+    int64_t seconds;
+    VELOX_CHECK(Timestamp::utcToEpoch(dateTime, seconds));
+    result = Timestamp(seconds, 0);
```

Comment:
how about using boost library? i'm not sure which one has better performance.
```
boost::posix_time::ptime time_utc(
    boost::gregorian::date(
        dateTime.tm_year + 1900, // tm_year is years since 1900
        dateTime.tm_mon + 1, // tm_mon is 0-based
        dateTime.tm_mday),
    boost::posix_time::hours(dateTime.tm_hour) +
        boost::posix_time::minutes(dateTime.tm_min) +
        boost::posix_time::seconds(dateTime.tm_sec));

auto time = boost::posix_time::to_time_t(time_utc);
result = Timestamp(time, 0);
```

Reply:
Boost is not known for its performance so I won't bet on that.  This function is simple and well defined, we'd rather have control over how it is implemented.